### ISSUE TITLE
Image size of the predicted mask and the truth mask are inconsistent

### ISSUE THREAD

**Issue by ax130885** [MedICL-VU/PRISM#2]
Hi, I found that the dataloader changes the data size to (1,1,128,128,128) and stores the prediction mask with the same size.
However, this size differs from MSD dataset,
which prevents me from overlaying the original image with the predicted mask in the 3D slicer to observe the tumor location.
How do you observe the prediction results?

Additionally, I noticed that the --save_csv parameter seems to have no effect.
I used find -name "*.csv" and couldn't find any files.
Maybe this is because tester.py is not used in inference, and the save_csv parameter is also not utilized in the Tester class of test.py.

**Comment by HaoLi12345** [comment]
Hi,

yes, for this current version, I forgot to change the correct order when saving the segmentations. simply change the order of x,y,z, e.g.  (B,C,x,y,z), would save this problem.

The --save_csv is disabled in this repo, please just ignore it.

**Comment by ax130885** [comment]
Sorry, I closed the issue accidentally and I don't quite understand what you mean.

My understanding is that you first resample the spcaing of images to 1,1,1 and then crop the patch size of 128,128,128 from resampled images.

If I want to restore it to its original shape, should I first place the patch to the correct position in the resampled image and then restore the spacing and image size?

**Comment by HaoLi12345** [comment]
yes, this is correct. the cropped position can be found by using tio.CropOrPad function in the dataloader file. the details can be found in torchio API.
or if you just want to see the qualitative results for accessing the performance, you can directly save the cropped image.

**Comment by ax130885** [comment]
I found some problems while reading the code.

I found that tio.croporpad in dataloader.py uses the center of the ground truth mask to crop the image.

Similarly, in get_next_click3D_torch_2() in test.py, the pred image and truth image will be compared to get FN and FP and serve as the prompt for the next iteration.

Does this mean that there is currently no way to make inferences from data without labels?

And I'm a little confused about whether humans must provide prompts for your project.
Because Human-in-loop is mentioned in the paper.
But the current program execution process does not require humans to provide prompts in each iteration.
Instead, it is provided by get_next_click3D_torch_2() mentioned above.

**Comment by HaoLi12345** [comment]
the current one uses the generated prompt, you can use the human-prompt instead of generated (skip get_next_click3D_torch_2()).

**Comment by ax130885** [comment]
But does label data still need to be the center of the crop during the inference process?

And if we want human interaction.
Do these following functions still need to be completed?
1. Output the result of each iter
2. Wait for the expert to enter prompt (human-machine interface may be required?)

**Comment by HaoLi12345** [comment]
The crop can be ignored based on different computation resource and specific disease. For current code, it is enough to cover the tumor area. The model is trained in this way, so the inference would have better performance in the same way. for other specific tasks, you can decide the train/inference strategy.
1. yes, you need to visualize the output to provide the corresponding prompt.
2. yes, this is ideal case. this is similar to the human annotation process.

**Comment by ax130885** [comment]
I think it might be unreasonable to use the current test crop method.
Because label data is required to test, does it not meet the purpose of prediction?

Perhaps it is a strategy to use a sliding window to perform inference on each block and finally merge all the inference results?
Sorry I'm not familiar with this area and don't know how this problem is usually solved.
Do you have any plan on how to solve this problem?

**Comment by HaoLi12345** [comment]
the current code mimics human behavior. For real human interaction, you might have a bounding box provided by human and crop the image based on it for efficient computation. The strategy is based on your task, some of tasks could be handled by crop and some of tasks could be handled by sliding window, it depends.
